packages/block-library/src/table/deprecated.js

```diff
@@ -6,13 +6,261 @@ import classnames from 'classnames';
 /**
  * WordPress dependencies
  */
-import { RichText, getColorClassName } from '@wordpress/block-editor';
+import {
+	RichText,
+	getColorClassName,
+	useBlockProps,
+} from '@wordpress/block-editor';
 
 const supports = {
 	align: true,
 };
 
+// As the previous arbitrary colors won't match theme color palettes, the hex
+// value will be mapped to the style.color.background attribute as if it was
+// a custom color selection.
+const oldColors = {
```

Comment:
> We are making every table block that exists use inline styles instead of a color class. But we will still need to keep the old classes so the blocks properly work on the frontend even if they are never opened on the editor.

If the block's are not opened in the editor and therefore not migrated. They would still keep the old CSS classes. This is why those classes still exist within the `style.css` file. 

> Could we leave the attributes as they were instead (keeping the classes and named attributes)? I guess we could keep the old colors as the default table block palette on lib/experimental-default-theme.json. Given that the classes still exist and will always be preset I guess the blocks should still look ok without any migration.

If the table block is given its own custom color palette, as you suggested, with the old colors under the same names, their CSS classes should be unchanged. With that approach, the original `backgroundColor` attribute could remain as is, instead of migrating it to the `style.color.background` attribute.